src/snowflake/cli/plugins/workspace/manager.py

```diff
@@ -0,0 +1,35 @@
+from pathlib import Path
+from typing import Dict
+
+from snowflake.cli.api.exceptions import InvalidProjectDefinitionVersion20Error
+from snowflake.cli.api.project.schemas.entities.entities import (
+    Entity,
+    v2_entity_model_to_entity_map,
+)
+from snowflake.cli.api.project.schemas.project_definition import DefinitionV20
+
+
+class WorkspaceManager:
+    def __init__(self, project_definition: DefinitionV20, project_root: Path):
+        if project_definition.definition_version != "2":
```

Comment:
we need a more robust version eventually though. Will the manager have to change for an eventual 2.1? I'd expect it usually wouldn't, which means that this check should really be "if major version == 2"